rr_common/include/rr_common/RelativePoseHistoryClient.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <ros/ros.h>
+#include <geometry_msgs/Pose2D.h>
+#include <nav_msgs/Path.h>
+
+
+namespace rr {
+
+class RelativePoseHistoryClient {
+public:
+  using Pose = geometry_msgs::Pose2D;
+
+  RelativePoseHistoryClient() = default;
+
+  /**
+   * Given a desired (recent) past time, get the interpolated pose at that time
+   * @param t past time
+   * @return Pose (x, y, theta) in current local frame
+   */
+  Pose GetRelativePoseAtTime(const ros::Time& t);
+
+  /**
+   * Register callback for /pose_history topic
+   * @param handle NodeHandle to use for subscription
+   */
+  ros::Subscriber RegisterCallback(ros::NodeHandle& handle);
```

Comment:
Makes more sense to register the callback in the constructor, then it's impossible to construct a RelativePoseHistoryClient that isn't subscribed to a topic

Reply:
future improvement